File: ToolShop_ListAndListener/src/main/java/com/example/redballtoy/toolshop_listandlistener/MainActivity.java
package com.example.redballtoy.toolshop_listandlistener;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private ListView listOfToolGoods;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //Убираем action bar
        /*
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.hide();
        }
        */


        listOfToolGoods = findViewById(R.id.lv_tools);
        //Добавим слушателя, в качестве объекта передадим анонимный класс
        listOfToolGoods.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                Toast.makeText(getApplicationContext(), String.valueOf(position), Toast.LENGTH_SHORT).show();

                Intent intent;
                switch (position) {
                    case 0:
                        intent = new Intent(getApplicationContext(), DrealCategoryActivity.class);
                        startActivity(intent);
                        break;
                    case 1:
                        break;
                    case 2:
                        break;
                }

            }
        });



    }
}

File: ListView_ToolsShop/src/main/java/com/example/redballtoy/listtest_toolsshop/MainActivity.kt
package com.example.redballtoy.listtest_toolsshop

import android.content.Intent
import android.os.Bundle
import android.widget.ListView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val actionBar = supportActionBar?.hide()

        val listViewTools: ListView = findViewById(R.id.lv_tools)
        listViewTools.setOnItemClickListener { parent, view, position, id ->
            Toast.makeText(this, "Нажали на позицию $position", Toast.LENGTH_SHORT).show()
            var intent: Intent
            when (position) {
                0 -> {
                    intent = Intent(this, DreelCategory::class.java).apply {}
                    startActivity(intent)
                }

                1 -> {
                }

                2 -> {
                }

            }


        }

    }
}

